lib/note-content-editor.tsx

```diff
@@ -350,7 +350,9 @@ class NoteContentEditor extends Component<Props> {
     if (
       this.editor &&
       this.state.editor === 'full' &&
-      prevProps.selectedSearchMatchIndex !== this.props.selectedSearchMatchIndex
+      prevProps.selectedSearchMatchIndex !==
+        this.props.selectedSearchMatchIndex &&
+      prevProps.noteId !== this.props.noteId
```

Comment:
If you performed a search then toggled through the search results and then performed a new search the focus would come back to the editor instead of finishing typing the search term.
With help from @codebykat I believe this was only intended for when a new note is selected so I added the additional check.

Reply:
This logic is for deciding when to reset the search selection, isn't it? What does it have to do with the editor focus?

I think the search selection should be updated whenever the search match index changes, not only when changing to a different note.

The actual problem here, which is perhaps being masked, seems to be that `setSearchSelection` makes another call to `this.focusEditor();`. That seems like something it should not do?

Reply:
You're right that is what this does. At first I didn't see any other time than note switches where this would be needed, but that was naive and I did of course find other times. 

You're right again that the problem stems from the call to `this.focusEditor();` in `setSearchSelection`. I had first thought to remove that all together but I'm sure especially if using the keyboard shortcuts to toggle through search matches you would want the focus to end up there so you could immediately edit the text.

I've changed my approach to this now. Instead of calling `this.focusEditor();` in `setSearchSelection` I've moved it to set focus in `setNextSearchSelection` and `setPrevSearchSelection`.

The only weird thing I've seen so far with this is that now the behavior is different if you are using the keyboard shortcuts to toggle through search results or if you are using the search results bar at the bottom to toggle through. If using the search results bar you don't get the focus anymore. The visual selection is still there though.

I think this may be OK as if you are using the keyboard shortcuts you wouldn't want to then have to click to edit, but if you are clicking the next and previous buttons it would likely be OK to click to where you want to edit.

Would love others thoughts on that part as well.

Reply:
This makes sense and I agree with you - I think leaving the focus on the buttons is best when using the buttons.